### PR TITLE
UX Refactor

### PR DESCRIPTION
**Pull request by amichard**
#221 
#222 
#223 
#224 
#225 

This pull request focuses mostly on the front-end. Below is a summary of the changes:
- Replaced Propose Trade link with button
- Added Credit Balance in the Credit Transactions page
- Added conditions on when to display the Propose Trade Button and tabs other than Credit Transactions
- Credit Transaction view details updates
  - Improvements on how price and number of credits is displayed
  - Improvements on the text being displayed so its more relevant to the type of transfer
- Minor refactoring with the routes
- Code clean-up for Lint

### CODE REVIEW

File: backend/api/serializers.py [hunk 1]
```diff
@@ -190,13 +190,26 @@ class Meta:
...
     class Meta:
         model = User
         fields = (
             'id', 'first_name', 'last_name', 'email', 'authorization_id',
             'authorization_guid', 'authorization_directory', 'display_name',
-            'organization')
+            'organization', 'organization_balance')
+
+    def get_organization_balance(self, obj):
+        try:
```
Comment (Sybrand) [inline backend/api/serializers.py]: Maybe a bit pedantic, but if this is not an exceptional circumstance, one shouldn't be handling it as an exception, but rather as a check. I would expect exception handling to be involved if something is actually wrong, and it be logged etc.

File: frontend/src/credit_transfers/components/CreditTransferTextRepresentation.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    this.totalValue = numeral(this.props.totalValue).format(NumberFormat.CURRENCY);
+    this.tradeEffectiveDate = (this.props.tradeEffectiveDate)
+      ? moment(this.props.tradeEffectiveDate).format('LL') : 'approval';
+
+    this.tradeStatus = (this.props.status.id === CREDIT_TRANSFER_STATUS.completed.id)
+      ? CREDIT_TRANSFER_STATUS.approved.description : this.props.status.status;
+  }
+
+  _buyAction () {
+    if (this.props.status.id === CREDIT_TRANSFER_STATUS.completed.id) {
```
Comment (Sybrand) [inline frontend/src/credit_transfers/components/CreditTransferTextRepresentation.js]: Just a personal preference note: I would do:
`return this.props.status.id === CREDIT_TRANSFER_STATUS.completed.id ? ' bought ' : ' is proposing to buy '`

I think it's easier to read, and your function only has one exit point.

File: frontend/styles/CreditTransfers.scss [hunk 3]
```diff
@@ -69,20 +64,22 @@
     
       .step {
         font-size: 14px;
+        font-weight: bold;
         text-align: center;
-        color: $light-font;
+        color: $medium-font;
```
Comment (Sybrand) [inline frontend/styles/CreditTransfers.scss]: 😀 woohoo SASS! Loving it. Try to guard against nesting too deeply though.

File: frontend/src/credit_transfers/components/CreditTransferTextRepresentation.js (hunk 2 above)
Comment (amichard) [reply frontend/src/credit_transfers/components/CreditTransferTextRepresentation.js]: sounds good. i have no problems with ternary functions.

**Pull request merged**